src/modules/Channel/context/hooks/useHandleReconnectForChannelList.ts

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect } from 'react';
+
+import {
+  GroupChannel,
+  GroupChannelListQuery,
+} from '@sendbird/chat/groupChannel';
+import { Logger } from '../../../../lib/SendbirdState';
+import useReconnectOnIdle from './useReconnectOnIdle';
+import { SdkStore } from '../../../../lib/types';
+import { ChannelListActionTypes } from '../../../ChannelList/dux/actionTypes';
+import { GroupChannelListQueryParamsInternal } from '../../../ChannelList/context/ChannelListProvider';
+import { MarkAsDeliveredSchedulerType } from '../../../../lib/hooks/useMarkAsDeliveredScheduler';
+import * as channelActions from '../../../ChannelList/dux/actionTypes';
+import { createChannelListQuery } from '../../../ChannelList/utils';
+import { DELIVERY_RECEIPT } from '../../../../utils/consts';
+
+interface UseHandleReconnectForChannelListProps {
+  // Dynamic props
+  isOnline: boolean;
+  reconnectOnIdle: boolean;
+
+  // Static props
+  logger: Logger;
+  sdk: SdkStore['sdk'];
+  currentGroupChannel: GroupChannel;
+  channelListDispatcher: React.Dispatch<ChannelListActionTypes>
+  setChannelSource: (query: GroupChannelListQuery) => void;
+  userFilledChannelListQuery: GroupChannelListQueryParamsInternal;
+  sortChannelList: (channels: GroupChannel[]) => GroupChannel[];
+  disableAutoSelect: boolean;
+  markAsDeliveredScheduler: MarkAsDeliveredSchedulerType;
+  disableMarkAsDelivered: boolean;
+}
+
+function useHandleReconnectForChannelList({
+  isOnline,
+  reconnectOnIdle,
+  logger,
+  sdk,
+  currentGroupChannel,
+  channelListDispatcher,
+  setChannelSource,
+  userFilledChannelListQuery,
+  sortChannelList,
+  disableAutoSelect,
+  markAsDeliveredScheduler,
+  disableMarkAsDelivered,
+}: UseHandleReconnectForChannelListProps): void {
+  const { shouldReconnect } = useReconnectOnIdle(isOnline, currentGroupChannel, reconnectOnIdle);
+
+  useEffect(() => {
+    return () => {
+      // state changed from offline to online AND tab is visible
+      if (shouldReconnect) {
+        logger.info('ChannelList - creating query', { userFilledChannelListQuery });
```

Comment:
Can we use a more specific description for this hook?
ex. 
```
logger.info('useHandleReconnecForChannelList: Start creating query, {...});
```

Reply:
Thanks for the catch I updated